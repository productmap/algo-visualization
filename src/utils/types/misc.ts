import { ElementStates } from "./element-states";

export type TLettersArray = {
  letter: string;
  state: ElementStates;
};

export type TColumn = {
  value: number;
  state: ElementStates;
};

export type TItem = {
  value: string;
  state: ElementStates;
};

export interface IQueue<T> {
  enqueue: (item: T) => void;
  dequeue: () => void;
  clear: () => void;
}

export type TCircle = {
  state: ElementStates;
  value?: string | null;
  add?: boolean;
  delete?: boolean;
  smallCircle?: {
    value?: string | null;
  };
};