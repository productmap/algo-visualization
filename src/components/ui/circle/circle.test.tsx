import renderer from "react-test-renderer";
import { Circle } from "./circle";
import { ElementStates } from "../../../utils/types/element-states";

it("Круг без буквы отрисовывается корректно", () => {
  const component = renderer.create(<Circle />).toJSON();
  expect(component).toMatchSnapshot();
});

it("Круг с буквой отрисовывается корректно", () => {
  const component = renderer.create(<Circle letter="A" />).toJSON();
  expect(component).toMatchSnapshot();
});

it("Круг с head отрисовывается корректно", () => {
  const component = renderer.create(<Circle head="head" />).toJSON();
  expect(component).toMatchSnapshot();
});

it("Круг с React в head отрисовывается корректно", () => {
  const component = renderer.create(<Circle head={<p>head</p>} />).toJSON();
  expect(component).toMatchSnapshot();
});

it("Круг с tail отрисовывается корректно", () => {
  const component = renderer
    .create(<Circle tail="tail" tailType="string" />)
    .toJSON();
  expect(component).toMatchSnapshot();
});

it("Круг с react в tail отрисовывается корректно", () => {
  const component = renderer
    .create(<Circle tail={<p>tail</p>} tailType="element" />)
    .toJSON();
  expect(component).toMatchSnapshot();
});

it("Круг с index отрисовывается корректно", () => {
  const component = renderer.create(<Circle index={1} />).toJSON();
  expect(component).toMatchSnapshot();
});

it("Круг с пропом isSmall ===  true", () => {
  const component = renderer.create(<Circle isSmall />).toJSON();
  expect(component).toMatchSnapshot();
});

it("Круг с состоянием Default отрисовывается корректно", () => {
  const component = renderer
    .create(<Circle state={ElementStates.Default} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});

it("Круг с состоянием отрисовывается корректно", () => {
  const component = renderer
    .create(<Circle state={ElementStates.Changing} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});

it("Круг с состоянием Modified отрисовывается корректно", () => {
  const component = renderer
    .create(<Circle state={ElementStates.Modified} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});
