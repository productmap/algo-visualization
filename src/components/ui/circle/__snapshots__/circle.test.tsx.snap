// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Круг без буквы отрисовывается корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Круг с React в head отрисовывается корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <p>
      head
    </p>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Круг с head отрисовывается корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Круг с index отрисовывается корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Круг с react в tail отрисовывается корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <p>
      tail
    </p>
  </div>
</div>
`;

exports[`Круг с tail отрисовывается корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Круг с буквой отрисовывается корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Круг с пропом isSmall ===  true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Круг с состоянием Default отрисовывается корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Круг с состоянием Modified отрисовывается корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Круг с состоянием отрисовывается корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
